dev/com.ibm.ws.jbatch.security/src/com/ibm/ws/jbatch/security/impl/WSBatchAuthServiceImpl.java

```diff
@@ -150,25 +175,84 @@ public long authorizedInstanceRead(long instanceId)
         throws JobSecurityException, NoSuchJobInstanceException {
 
         String submitter = persistenceManagerService.getJobInstanceSubmitter(instanceId);
+        Set<String> listOfGroupsForJobID = null;
+        Set<String> listOfGroupsForSubject = null;
+        boolean checkGroupSecurity = false;
 
         if (this.isAdmin(runAsSubject())) {
         } else if (this.isMonitor(runAsSubject())) {
```

Comment:
I think the nested if/else is complicated enough that you should just return out of here.